components/script/dom/serviceworkerglobalscope.rs

```diff
@@ -223,19 +231,22 @@ impl ServiceWorkerGlobalScope {
                 data.read(GlobalRef::Worker(scope), message.handle_mut());
                 MessageEvent::dispatch_jsval(target, GlobalRef::Worker(scope), message.handle());
             },
-            WorkerScriptMsg::Common(CommonScriptMsg::RunnableMsg(_, runnable)) => {
+            ServiceWorkerScriptMsg::CommonWorker(WorkerScriptMsg::Common(CommonScriptMsg::RunnableMsg(_, runnable))) => {
                 runnable.handler()
             },
-            WorkerScriptMsg::Common(CommonScriptMsg::RefcountCleanup(addr)) => {
+            ServiceWorkerScriptMsg::CommonWorker(WorkerScriptMsg::Common(CommonScriptMsg::RefcountCleanup(addr))) => {
                 LiveDOMReferences::cleanup(addr);
             },
-            WorkerScriptMsg::Common(CommonScriptMsg::CollectReports(reports_chan)) => {
+            ServiceWorkerScriptMsg::CommonWorker(WorkerScriptMsg::Common(CommonScriptMsg::CollectReports(reports_chan))) => {
                 let scope = self.upcast::<WorkerGlobalScope>();
```

Comment:
@jdm I've made that change adding `CommonWorker`, but i think this becomes very nested in matching in `handle_script_event`. Also not sure to reformat the match arms as it exceeds tidy's line length check. Suggestions ?